azuredevops/internal/service/taskagent/data_variable_group.go

```diff
@@ -0,0 +1,125 @@
+package taskagent
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6/taskagent"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/client"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/tfhelper"
+)
+
+func DataVariableGroup() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceVariableGroupRead,
+		Schema: map[string]*schema.Schema{
+			vgProjectID: {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.IsUUID,
```

Comment:
API accept both `projectId` and `projectID`, `validateFunc` should not only validate the `project ID`

Reply:
Are you talking about the project name? The `azuredevops_variable_group` resource only accept `project_id` (even if documentation says otherwise....) and uses the same `ValidateFunc`...

This is kind of inconsistent throughout the provider, not sure what to do here...

Reply:
When I review the doc and saw the `project_id` support both `ID` and `name` so I rechecked the API.  It is enough to support only `project_id`, the documentation needs to be updated.

Reply:
Documentation updated!